Response to reviewers (#16)
greenelab/word_lapse_manuscript: content/09.response-to-reviewers.md

# Response to Reviewers

## Reviewer 1

>In this paper, the authors presented a method to examine the semantic shift in biomedical preprints and pre-publication by machine learning method. This work is very impressive and could have a great impact on the domain. However, the following questions should be addressed before publication.

>Q1: In this work, the author applied word2vec model to train the word embeddings. The main issue is that word2vec has a limited capability to capture contextual information. I would suggest the authors  try other more advanced language model, like BERT to train embeddings.

**At the time of this work, we found that Word2Vec was easier to interpret than BERT.

> Given the unknown amount of true changepoints, we decided to move forward with Word2Vec

Reviewer: ```suggestion
Given the unknown amount of true changepoints and the lack of a consensus gold standard, we decided to move forward with Word2Vec so that we could benefit from the interpretability of the model.
```

> However, using BERT within this context is an area for future work.**

Reviewer: ```suggestion
However, we agree that using BERT or related language models is an important area for future work.**
```

>Q2: How does the method take care of the new word issue? basically, if there’s a word created in this domain for only a few years and there are very few publications talking about it, then how the performance of this method would be?

> **In this situation, a new word wouldn't be detected as you would need more papers over time to detect any differences.

Reviewer: ```suggestion
**In the most technical sense, a new word could be detected as having a changepoint for any potential string of subsequent years in which the word appeared in both corpora.
However, we think that most new words are unlikely to be detected as they settle into initial use because most new words will occur relatively few times in the literature.
This would lead to high intra-year variability, reducing the likelihood of a discovered changepoint (this is the intent of our correction for intra-year variability between models).
The case where a new word would have detectable changepoints would be limited to those where the use is sufficiently common that its positioning can be estimated reliably and where its positioning changes between subsequent years. 
```

Reviewer: COV2 shows that it is possible for a word to overcome low usage to have a changepoint, but it's likely going to have to be a huge shift to get there: https://greenelab.github.io/word-lapse/?search=cov2&year=2010

> Arguably, it is unlikely that a word would have a semantic change in this context.

Reviewer: ```suggestion
```

> As we found within our paper and previous works, a word needs to have significant usage for a change to occur.

Reviewer: ```suggestion
```

**

>Q3: When detecting semantic changes, the cosine distance was used for comparing similarities between words. This method seems too easy and not that robust. Are there any other good ways for detecting?

**New ways of detecting semantic change remain open for investigation.

> However, most work that detects semantic changes uses cosine similarity (1-cosine distance).

Reviewer: ```suggestion
However, most work that detects semantic changes uses cosine similarity (1-cosine distance).
We added the intra- and inter-year comparisons in part to improve the robustness of this metric, which is widely used within the field.
```

> Due to this work being the first within biomedical literature, we sought to keep the metric the same.

Reviewer: ```suggestion
Due to this work being the first within biomedical literature, we adapted the commonly used metric, but we agree with the reviewer that future work should explore additional metrics.
```

**

## Reviewer 2

>This paper develops a method to evaluate semantic shifts by calculating annual and interannual changes using multiple integrated models. This method achieves good results, but there are some shortcomings in the paper. There are some suggestions for revision.
>1.      The motivation is not clear. Please specify the importance of this paper.

**We have updated our introduction to make our motivation for this work more clear.**
```diff
```

>2.      Please highlight the contributions of this paper.

**We have updated our introduction to make our contributions for this work more clear.**
```diff
```

>3.      Most of references are out of date. Please discuss more recently published solutions, especially the solutions published in 2022.

**We have updated our introduction to include more recent publications that pertain to this work.**

>4.      In the method part, the authors’ description is not easy to understand. Have the authors considered using drawings to explain the method?

> **We have updated our methods section to include a graphical depiction of our data pipeline.**

Reviewer: ```suggestion
**We agree with the reviewer that this was a significant opportunity for improvement. We have updated our methods section to include a graphical depiction of our data pipeline, which is now Figure X [TODO: Fill in figure number].**
```

>5.      When detecting semantic changes across time, the authors mention the use of Cartesian products in calculating interannual changes. What are the advantages of using Cartesian products for the task of this paper?

> **The main advantage here is that the cartesian product allows us to directly compute the inter-year distances without the need for a statistical correction.

Reviewer: ```suggestion
**The main advantage here is that the cartesian product allows us to directly compute the inter-year distances.
```

> Since we compute all possible combinations of model pairings, our sample size is sufficiently large enough to obtain inter-year distance estimates.**

Reviewer: ```suggestion
Since we compute all possible combinations of model pairings, our sample size is sufficiently large enough to obtain reliable inter-year distance estimates for the years studied (noting that we removed pre-2000 literature due to the limited size of PubMed in these years).**
```

>6.      There is something unclear about Figure 1. There are five black dots in the second picture of Figure 1. What is the specific meaning of these five black dots? Please explain it to the author.

> **The black dots in Figure 1 are the individual word vectors obtained from their corresponding word2vec model.

Reviewer: for which word? "Probiotics"? If so, I recommend adding that here.

Reviewer: Also a good point to make a nod to your new figure which should make this process a lot clearer!

We greyed out all tokens except for 'probiotics' to provide an individual example on what occurs without aligning models.**

>7.      Figure 2 has some ambiguities. The authors use single model and multiple model to carry out experiments, and the experimental results are different. Other than using different models, what could be the reason for the difference?

**One reason for the differences is that our correction metric allows for different scales of differences.
Since we are taking a ratio of the inter-year distances over the intra-year distances, we encounter values that are greater than 0.

> Ultimately, the point for figure 2 is to show that the confidence intervals are a lot smaller when you correct for variability than not.

Reviewer: ```suggestion
Ultimately, the point for figure 2 is to show that the confidence intervals are a lot smaller when variability is accounted for than not.
```

**

>8.      For the proposed method, has the author considered applying it to the same type of task and achieving similar results as in this paper? Please explain it to the readers.

**We have thought of using this approach outside of biomedical literature.
However, we determined that this direction would be outside our paper's scope.

> We did mention this point as a future direction within our manuscript.**

Reviewer: ```suggestion
We mention this point as a future direction within our manuscript.**
```

>9.      More discussions of technical details should be given.

> **We have updated our manuscript to make our technical details more clear.**

Reviewer: ```suggestion
**We have updated our manuscript to provide additional discussion of the underlying technical details.**
```

>•       In the introduction, what key theoretical perspectives and empirical findings in the main literature have already informed the problem formulation? What major, unaddressed puzzle, controversy, or paradox does this research address?

**We have updated our introduction to include more recent publications and discussion in relation to this work.**

>•       Authors should further clarify and elaborate novelty in their contribution.

**We have updated our introduction to make our contributions for this work more clear.**
```diff
```

>•       What are the limitations of the present work?

**One limitation of this work is that changepoint validation can be challenging.

> Due to this being the first time this work has been performed, there isn't an available gold set.

Reviewer: ```suggestion
Due to this being the first time this work has been performed, there isn't an available widely-agreed upon gold standard set.
```

Our solution to circumvent this problem is that we provide a website that allows users to investigate our changepoint list further.
**

## Reviewer 4

>This is a well-written manuscript to examine semantic shift in open access biomedical preprints and pre-publication peer-reviewed text. The methods are novel and  are clearly described. The results are clearly presented. It is a pleasure and very easy for audience to follow the paper. It adds to scientific value of the relative research field. I suggest acceptance.

**We appreciate the positive feedback from the reviewer.**

## Reviewer 5

>Dear authors,
>It is a pleasure to review your manuscript.

**We appreciate the positive feedback from the reviewer.**

>My suggestion:
>1. Thoroughly revise the manuscript.

> **We have thoroughly revised our manuscript based on the feedback that was given.**

Reviewer: ```suggestion
**We have thoroughly revised our manuscript based on the feedback that was given by the other reviewers and based on an additional proof-read of our work.**
```

>2. Visit the Submission Guidelines and place your manuscript according to the journal's guidelines.
>Preparing main manuscript text
>Preparing illustrations and figures
>Preparing tables
>Preparing additional files

> **We have examined the journal guidelines and have made sure everything is correct.**

Reviewer: ```suggestion
**We have examined the journal guidelines and updated formatting.**
```

>3. The figures must be well displayed, the text of the figures must be legible.

**We updated our figures to meet the journal's requirements.**

>4. The introduction has few quotes, it is very short.

**We have extended our introduction to include a more thorough discussion on previous work and highlight our contrubtions/motivation.**

>5. The methodology is not well detailed, there is no figure that represents the pipeline.

**We have updated our methods section to include a graphical depiction of our data pipeline.**

## Reviewer 6

>Reviewer Comments
>Manuscript Number: Not Mentioned
>The topic is exciting and shows how the words are changing over time. However, the authors must consider the following comments to improve the manuscript's quality.
>
>1.      The authors can use a graphical representation of the proposed work.

**We have updated our methods section to include a graphical depiction of our data pipeline.**

>2.      The authors must explain how the methodology is unique.

> **Our work is the first example where year variability has been accounted for within a word2vec model.**

Reviewer: ```suggestion
**Our work is the first example where year variability has been accounted for within a word2vec model. The revised figure helps to demonstrate this approach, which previously was somewhat buried in the technical details of the methods section.**
```

>3.      The resolution of the Figures can be improved for better visibility.

**We updated our figures to meet the journal's requirements.**

>4.      The authors should discuss the computational complexity of the methods.

> **We refrain from including this information as it appears outside of our paper scope.**

Reviewer: ```suggestion
**As with work focused on the training of machine learning models, it is difficult to estimate this precisely. Because we add inter- and intra-year variability, we note that the distance caluclations scale with the square of the number of models used per year and the number of years examined (if all combinations are calculated, as they are for our manuscript to examine stability over time); however, more efficiency could be gained in subsequent work by only examining subsequent year pairs now that the statistic has been evaluated.**
```

# Response to Reviewers

## Reviewer 1

>In this paper, the authors presented a method to examine the semantic shift in biomedical preprints and pre-publication by machine learning method. This work is very impressive and could have a great impact on the domain. However, the following questions should be addressed before publication.

>Q1: In this work, the author applied word2vec model to train the word embeddings. The main issue is that word2vec has a limited capability to capture contextual information. I would suggest the authors  try other more advanced language model, like BERT to train embeddings.

**At the time of this work, we found that Word2Vec was easier to interpret than BERT.

> Given the unknown amount of true changepoints and the lack of a consensus gold standard, we decided to move forward with Word2Vec to benefit from the model's interpretability.

Reviewer: ```suggestion
Given that true changepoints are unknown and the lack of a consensus gold standard, we decided to move forward with Word2Vec to benefit from the model's interpretability.
```
